### PR TITLE
add reference to regulargridinterpolator

### PR DESCRIPTION
**Pull request by lhoupert**
#### Reference issue
Issue https://github.com/scipy/scipy/issues/16492

#### What does this implement/fix?
Add a reference to RegularGridInterpolator in interp2d and recommendation to use it when coordinates are not in ascending order

### CODE REVIEW

File: scipy/interpolate/_interpolate.py [hunk 1]
```diff
@@ -165,6 +165,8 @@ class interp2d:
         Spline interpolation based on FITPACK
     BivariateSpline : a more recent wrapper of the FITPACK routines
     interp1d : 1-D version of this function
+    RegularGridInterpolator : interpolation on a regular or rectilinear grid
+        in arbitrary dimensions (`interpn` wraps this class).
```
Comment (ev-br) [inline scipy/interpolate/_interpolate.py:169]: Nit: Better add a separate heading for `interpn`

File: scipy/interpolate/_interpolate.py [hunk 2]
```diff
@@ -176,8 +178,9 @@ class interp2d:
     of 0. If more control over smoothing is needed, `bisplrep` should be
     used directly.
 
-    The new data points to interpolate `xnew` and `ynew` have to be
-    sorted by increasing order.
+    The coordinates of the data points to interpolate `xnew` and `ynew`
+    have to be sorted by ascending order. If it is not the case
+    `RegularGridInterpolator` should be used instead.
```
Comment (ev-br) [inline scipy/interpolate/_interpolate.py:183]: While we're at it, let's strengthen the message: `interp2d` is legacy and is not recommended for use in new code. New code should use `RegularGridInterpolator` instead.

**Review by ev-br — CHANGES_REQUESTED** [review]
While we're at it, let's actually strengthen the message and say it explicitly that `interp2d` is legacy and refer users to RGI.

File: scipy/interpolate/_interpolate.py (hunk 1 above)
Comment (lhoupert) [reply scipy/interpolate/_interpolate.py]: Should be resolve by https://github.com/scipy/scipy/pull/16511/commits/52d52a16b083b5552fa7084ea0e02cd86e2b89a7

File: scipy/interpolate/_interpolate.py (hunk 2 above)
Comment (lhoupert) [reply scipy/interpolate/_interpolate.py]: should be resolved by https://github.com/scipy/scipy/pull/16511/commits/a50b21476e97a0e2b809d9c777eef0cce2459c5c

**Comment by ev-br** [comment]
Thank you @lhoupert . I'll leave it open for a while, to if there are further comments: these sorts of recommendations are not to be taken lightly.

**Review by AtsushiSakai — APPROVED** [review]
I'm +1 to recommend using `RegularGridInterpolator` instead of `interp2d`.

File: scipy/interpolate/_interpolate.py [hunk 3]
```diff
@@ -176,8 +180,10 @@ class interp2d:
     of 0. If more control over smoothing is needed, `bisplrep` should be
     used directly.
 
-    The new data points to interpolate `xnew` and `ynew` have to be
-    sorted by increasing order.
+    The coordinates of the data points to interpolate `xnew` and `ynew`
+    have to be sorted by ascending order. interp2d is legacy and is not
```
Comment (ev-br) [inline scipy/interpolate/_interpolate.py:184]:
```suggestion
    have to be sorted by ascending order. 
    
    `interp2d` is legacy and is not
```

File: scipy/interpolate/_interpolate.py [hunk 4]
```diff
@@ -176,8 +180,12 @@ class interp2d:
     of 0. If more control over smoothing is needed, `bisplrep` should be
     used directly.
 
-    The new data points to interpolate `xnew` and `ynew` have to be
-    sorted by increasing order.
+    The coordinates of the data points to interpolate `xnew` and `ynew`
+    have to be sorted by ascending order.
```
Comment (ev-br) [inline scipy/interpolate/_interpolate.py:184]:
```suggestion
    have to be sorted by ascending order.
```

File: scipy/interpolate/_interpolate.py [hunk 5]
```diff
@@ -176,8 +180,12 @@ class interp2d:
     of 0. If more control over smoothing is needed, `bisplrep` should be
     used directly.
 
-    The new data points to interpolate `xnew` and `ynew` have to be
-    sorted by increasing order.
+    The coordinates of the data points to interpolate `xnew` and `ynew`
+    have to be sorted by ascending order. 
+
```
Comment (ev-br) [inline scipy/interpolate/_interpolate.py:185]:
```suggestion

```

**Review by ev-br — COMMENTED** [review]
rm whitespace so that the linter does not complain

**Review by ev-br — APPROVED** [review]

**Comment by ev-br** [comment]
OK, this has been approved by a core dev, there are no further comments, let's merge this now. Thanks @lhoupert , keep them coming!

**Pull request merged**